functions/Get-DbaAgentAlert.ps1

```diff
@@ -71,17 +84,33 @@ function Get-DbaAgentAlert {
 
             $alerts = $server.Jobserver.Alerts
 
-            foreach ($alert in $alerts) {
-                $lastraised = [dbadatetime]$alert.LastOccurrenceDate
+            if (Test-Bound 'Alert') {
+                $tempAlerts = @();
+
+                foreach ($a in $Alert) {
+                    $tempAlerts += $alerts | where Name -like $a;
```

Comment:
Instead of iterating over them twice...would it not be simpler to just but this filter in the output block?

Use of the `+=` adds memory usage if user is hitting a server that has a ton of alerting setup.

Reply:
Hope you dont mind, I will merge as the fix works and passes tests (tests that failed are unrelaated)

Reply:
Our test pass because we are not testing anything specific for this command. The integration test for this command does not do anything other than making sure the unfiltered output includes an alert we created in Appveyor instance.

Reply:
ohh nooo

Reply:
I'm happy to both add to the tests and write some better tests. But under separate cover (since this PR is about to have a two month birthday... they grow up so fast!).

Can either of you suggest a cmdlet that currently has decent tests that I can emulate?

Reply:
It's already been merged so you have to do a new PR 😉

Reply:
Likely `Get-DbaDatabase` is pretty solid test, may have more than what is needed for this but shows how to mock some internal commands (though should not be needed for this one). Main thing is just checking the code coverage to make sure at least some portion of code for the parameters is being executed. End result of a test won't necessarily get 100% coverage but the base functionality the command should have, we want to test for and make sure it returns good....especially areas of a command we have "special" logic.

Reply:
Roger that on "separate PR". And thanks for pointing me in a direction. I'll see what I can do!